API.php

```diff
@@ -1477,7 +1480,15 @@ public function importContainerVersion($exportedContainerVersion, $idSite, $idCo
 
         $this->containers->checkContainerVersionExists($idSite, $idContainer, $idContainerVersion);
         $this->enableGeneratePreview = false;
-        $this->import->importContainerVersion($exportedContainerVersion, $idSite, $idContainer, $idContainerVersion);
+        try {
+            $this->import->importContainerVersion($exportedContainerVersion, $idSite, $idContainer, $idContainerVersion);
+        } catch (Exception $e) {
+            if (!$isDraftRestoreCall && !empty($draft)) {
+                // rollback to old working draft
```

Comment:
Would it be cleaner to delete the backup version created on line 1478? It feels confusing that a new version is created each time even if the import fails. You could simply add `$backupVersionId = ` to line 1478 and do the following:
```suggestion
            if (!$isDraftRestoreCall && !empty($draft)) {
                // Delete the backup since the import failed and is being rolled back
                if (!empty($backupName)) {
                    $this->deleteContainerVersion($idSite, $idContainer, $backupVersionId);
                }
            
                // rollback to old working draft
```

Reply:
Done :+1: